src/main/java/com/based/model/StupidStorage.java

```diff
@@ -54,4 +45,133 @@ public List<Row> getRows(int[] columns) {
+                }
+            }
+        }
+        // if(aggregates != null){
+        //     for(Aggregate agg : aggregates){
+        //         if(agg.getFunction().equals("count")){
+        //             int count = filteredRows.size();
+        //             List<Object> newList = new ArrayList<>();
+        //             newList.add(count);
+        //             //add only the first row to the list output 
+        //             if(count > 0){
+        //                 System.err.println("response.size() > 0");
+        //                 for(Object o : filteredRows.get(0).getValues()){
+        //                     newList.add(o);
+        //                 }
+        //             }
+        //             filteredRows.clear();
+        //             filteredRows.add(new Row(newList));
+        //         }
+        //     }
+        // }
+
+        if(aggregates != null && !aggregates.isEmpty()){
+            int count = filteredRows.size();
+            List<Object> newList = new ArrayList<>();
+            for(Aggregate agg : aggregates){
+                if(agg.getFunction().equals("count")){
+                    newList.add(count);                    
+                }
+                else if(agg.getFunction().equals("sum")){
+                    String target = agg.getColumn_target();
+                    int targetIndexe = getRowsCallback.getTargetIndexe(target, columns);
+                    Object sum = 0;
+
+                    for(Row row : filteredRows){
+                        List<Object> rowValues = row.getValues();
+                        if(rowValues.size() > 0){
+                            if(rowValues.get(targetIndexe) instanceof Integer){
+                                sum = (Integer) sum + (Integer) rowValues.get(targetIndexe);
+                            }
+                            else if(rowValues.get(targetIndexe) instanceof Float){
+                                sum = (Float) sum + (Float) rowValues.get(targetIndexe);
+                            }
+                            else {
+                                throw new Exception("Can sum only integer or float types");
+                            }
+                        }
+                    }
+                    newList.add(sum);
+                }
+            }
+            //add only the first row to the list output
+            if(count > 0){
+                for(Object o : filteredRows.get(0).getValues()){
+                    newList.add(o);
+                }
+            }
+            filteredRows.clear();
+            filteredRows.add(new Row(newList));
+        }
+        return filteredRows;
+    }
+
+   
+    /**
+     * wherePredicate can be null
+     */
+    @Override
+    public HashMap<String, List<Row>> groupByFilter(Predicate<Row> wherePredicate, int[] columns, int groupby) throws Exception {
+        HashMap<String, List<Row>> map = new HashMap<>();
```

Comment:
Juste une petite suggestion par habitude
```suggestion
        Map<String, List<Row>> map = new HashMap<>();
```